_posts/2019-12-25-technical-debt.md

```diff
@@ -0,0 +1,107 @@
+
+Furthermore, this brings to mind how we would utilise these metrics in a system that will continue to be used in the forseeable future but isn't undergoing rapid change and remains stable. 
+
+- Lead Time For Changes would trend towards zero; no changes means no lead time.
+- Change Failure Rate would trend towards zero; no changes mean that no changes can fail.
+- Deployment Frequency would trend towards zero; no changes (often) mean no deployments.
+- Time to Restore Service; assuming system is indeed stable (humor this), this would also trend to zero as the system would never excercise the need to be restored.
+
+A perfect system would therefore trend to zero on all measures but we know such a system does not exist; all systems fail at some point, resulting in a need to restore a service. This would likely result in a change and therefore effect all the other metrics. This does mean that systems that are not underdoing rapid change (or failing often) can harbour serious technical debt. There is an inherent risk that a major outage could occur that is difficult to repair. In younger systems deployment is an inherently chaotic event that affords the opportunity to impact all the metrics and therefore discover opportunity for improvement. This opportunity is not as actively present in maturer systems and has to be introduced. Chaos engineering therefore becomes more important as systems approach a certain age and criticality to the business; it introduces a kind of controlled change that normally only occurs during active development. In saying this it may be appropriate for teams starting out with chaos engineering to regularly schedule the deployment of code to actively test the ability to deploy frequently even if no changes have been made.
+
+The previous paragraph addresses the technical debt problem for existing systems but does not give any insight into how we can actively manage technical debt for new and actively developed systems. We would like to answer the question, "Are there any tactics that software departments can employ in order to manage the risks of technical debt?". We can see two common solutions to this problem via the Google "site reliability engineering" and AWS "two pizza team" models.
```

Comment:
Split this para I think. Pose the question in the first para, and then have the answer as a one sentence paragraph after.